spring-ai-core/src/main/java/org/springframework/ai/chat/client/observation/DefaultChatClientObservationConvention.java

```diff
@@ -0,0 +1,144 @@
+/*
+* Copyright 2024 - 2024 the original author or authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.springframework.ai.chat.client.observation;
+
+import java.util.stream.Collectors;
+
+import org.springframework.ai.chat.client.observation.ChatClientObservationDocumentation.LowCardinalityKeyNames;
+import org.springframework.ai.chat.observation.ChatModelObservationDocumentation;
+import org.springframework.lang.Nullable;
+import org.springframework.util.CollectionUtils;
+
+import io.micrometer.common.KeyValue;
+import io.micrometer.common.KeyValues;
+
+/**
+ * @author Christian Tzolov
+ * @since 1.0.0
+ */
+
+public class DefaultChatClientObservationConvention implements ChatClientObservationConvention {
+
+	public static final String DEFAULT_NAME = "gen_ai.chat.client.operation";
+
+	private static final KeyValue CHAT_CLIENT_TOOL_FUNCTION_CALLBACKS_NONE = KeyValue.of(
+			ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_TOOL_FUNCTION_CALLBACKS,
+			KeyValue.NONE_VALUE);
+
+	private static final KeyValue CHAT_CLIENT_TOOL_FUNCTION_NAMES_NONE = KeyValue.of(
+			ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_TOOL_FUNCTION_NAMES,
+			KeyValue.NONE_VALUE);
+
+	private static final KeyValue CHAT_CLIENT_ADVISOR_NONE = KeyValue
+		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_ADVISORS, KeyValue.NONE_VALUE);
+
+	private static final KeyValue CHAT_CLIENT_ADVISOR_PARAM_NONE = KeyValue
+		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_ADVISOR_PARAMS, KeyValue.NONE_VALUE);
+
+	private final String name;
+
+	public DefaultChatClientObservationConvention() {
+		this(DEFAULT_NAME);
+	}
+
+	public DefaultChatClientObservationConvention(String name) {
+		this.name = name;
+	}
+
+	@Override
+	public String getName() {
+		return this.name;
+	}
+
+	@Override
+	@Nullable
+	public String getContextualName(ChatClientObservationContext context) {
+		return "%s %s".formatted(context.getOperationMetadata().operationType(),
```

Comment:
Since the framework operations are at a higher abstraction level than the OTel semantic conventions, I would suggest to go with our own naming strategy here. Otherwise, all Spring AI own operations would all have the same span name (`framework spring_ai`).

I would suggest a naming strategy such as `spring_ai chat_client` (the first value being the "provider", the second value being the "Spring AI Kind") perhaps even normalised without the underscore: `spring ai chat client`.

Other frameworks like OpenLit do something similar.